connaisseur/validators/cosign/cosign_validator.py

```diff
@@ -43,25 +41,25 @@ async def validate(
             "threshold",
             1 if trust_root != "*" or any(required) else len(self.trust_roots),
         )
-
-        self.vals = self.__get_pinned_keys(trust_root, required, threshold)
+        # vals is a validations dict for each required trust root containing validated digests and errors
+        vals = self.__get_pinned_keys(trust_root, required, threshold)
 
         # use concurrent.futures for now
         # tasks = [self.__validation_task(k, str(image)) for k in self.vals.keys()]
         # await asyncio.gather(*tasks)
 
         # prepare executor
-        num_workers = len(self.vals)
+        num_workers = len(vals)
         executor = ThreadPoolExecutor(num_workers)
         # prepare tasks
-        arguments = [(k, str(image)) for k in self.vals.keys()]
+        arguments = [(k, v.copy(), str(image)) for k, v in vals.items()]
```

Comment:
I'd add a comment there to justify the usage of v.copy as I wouldn't say its obvious. May help someone in the future touching this

Reply:
Will add a comment